stmhal/sdcard.c

```diff
@@ -96,6 +96,27 @@
 
 #endif
 
+//If no custom SDIO pins defined, use the default ones
+#ifndef MICROPY_HW_SDMMC_D0
+    #define MICROPY_HW_SDMMC_D0 (pin_C8)
+#endif
+#ifndef MICROPY_HW_SDMMC_D1
+    #define MICROPY_HW_SDMMC_D1 (pin_C9)
+#endif
+#ifndef MICROPY_HW_SDMMC_D2
+    #define MICROPY_HW_SDMMC_D2 (pin_C10)
+#endif
+#ifndef MICROPY_HW_SDMMC_D3
+    #define MICROPY_HW_SDMMC_D3 (pin_C11)
+#endif
+#ifndef MICROPY_HW_SDMMC_CK
+    #define MICROPY_HW_SDMMC_CK (pin_C12)
+#endif
+#ifndef MICROPY_HW_SDMMC_CMD
+    #define MICROPY_HW_SDMMC_CMD (pin_D2)
+#endif
```

Comment:
For this auto-config it would be simpler to just check if MICROPY_HW_SDMMC_CK is defined or not.  If not, then provide all the default definitions.  If it is, then don't provide any.  If the user is going to provide custom config values then they most likely will provide all of them, not a subset (this matches how it's done for some other peripherals).